### PR TITLE
Fix redundant requests for realm info

### PR DESCRIPTION
**Pull request by FadhlanR**
Ticket: CS-6761

There was a redundant request for the same realm info because we only cached the realm info value. As suggested in the ticket, I implemented caching for the promise.

### CODE REVIEW

File: packages/host/app/services/realm-info-service.ts [hunk 1]
```diff
@@ -138,4 +138,15 @@ export default class RealmInfoService extends Service {
       waiter.endAsync(token);
     }
   });
+
+  private fetch(url: string, init: RequestInit) {
```
Comment (jurgenwerk) [inline packages/host/app/services/realm-info-service.ts]: `cachedFetch` maybe?

File: packages/host/tests/integration/realm-info-service-test.gts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+      }
+
+      return null;
+    };
+    await Promise.all([
+      realmInfoService.fetchRealmInfo({ realmURL: new URL(testRealmURL) }),
+      realmInfoService.fetchRealmInfo({ realmURL: new URL(testRealmURL) }),
+      realmInfoService.fetchRealmInfo({ realmURL: new URL(testRealmURL) }),
+    ]);
+    assert.strictEqual(totalRealmInfoRequest, 1);
```
Comment (jurgenwerk) [inline packages/host/tests/integration/realm-info-service-test.gts:162]: nice

**Review by jurgenwerk — APPROVED** [review]

**Review by tintinthong — APPROVED** [review]

**Review by burieberry — APPROVED** [review]

File: packages/host/app/services/realm-info-service.ts (hunk 1 above)
Comment (habdelra) [reply packages/host/app/services/realm-info-service.ts]: Let’s type the return value for this method so it’s clear that it is returning a promise even though it’s a synchronous function.

File: packages/host/app/services/realm-info-service.ts [hunk 3]
```diff
@@ -99,12 +100,11 @@ export default class RealmInfoService extends Service {
...
-        let realmInfoResponse = await this.loaderService.loader.fetch(
-          `${realmURLString}_info`,
-          { headers: { Accept: SupportedMimeType.RealmInfo } },
-        );
+        let realmInfoResponse = await this.fetch(`${realmURLString}_info`, {
+          headers: { Accept: SupportedMimeType.RealmInfo },
+        });
 
-        let realmInfo = (await realmInfoResponse.json())?.data
+        let realmInfo = (await realmInfoResponse.clone().json())?.data
```
Comment (habdelra) [inline packages/host/app/services/realm-info-service.ts:111]: I’m curious, why do we clone this?

Comment (FadhlanR) [reply packages/host/app/services/realm-info-service.ts:111]: Avoiding the `json body is already read` error. I'll add a comment.

Comment (FadhlanR) [reply packages/host/app/services/realm-info-service.ts:111]: Added [1e1da51](https://github.com/cardstack/boxel/pull/1259/commits/1e1da5101b8bf2bc14d3f4d4f311443551b22329)

**Pull request merged**